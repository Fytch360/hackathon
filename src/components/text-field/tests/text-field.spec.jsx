import React from 'react'
import { shallow, mount } from 'enzyme'

import { TextField } from '..'

describe('<TextField />', () => {
    test('Is available', () => {
        expect(TextField).toBeDefined()
    })

    test('Snapshot', () => {
        const propsExample = {
            id: 'idValue',
            icon: 'iconExampleValue',
            placeholder: 'Текст',
            value: 'testValue',
            a11y: { label: 'Демонстрационное поле' },
            disabled: true,
            onClick: jest.fn(),
            onChange: jest.fn()
        }

        const wrapper = mount(<TextField {...propsExample} />)

        expect(wrapper).toMatchSnapshot()
    })

    test('Should trhow id for input', () => {
        const idValue = 'id-value-.-.'
        const wrapper = shallow(<TextField id={idValue} />).dive()

        expect(wrapper.children().first().prop('id')).toBe(idValue)
    })

    test('Should render additional text', () => {
        const additionalTextValue = 'Text that dreams of being main...'
        const wrapper = shallow(<TextField additionalText={additionalTextValue} />).dive()

        expect(wrapper.text()).toBe(additionalTextValue)
    })

    test('Should render additionalChild', () => {
        const additionalChildNode = <div id="additionalChild" />
        const wrapper = shallow(<TextField additionalChild={additionalChildNode} />).dive()

        expect(wrapper.contains(additionalChildNode)).toBeTruthy()
    })

    test('Should throw "disabled" prop for text input and additional child', () => {
        const additionalChildNode = <div id="additionalChild" />
        const wrapper = shallow(<TextField disabled additionalChild={additionalChildNode} />).dive()

        expect(wrapper.children().first().prop('disabled')).toBeTruthy()
        expect(wrapper.children().last().prop('disabled')).toBeTruthy()
    })

    test('Should throw "readonly" prop for additional child', () => {
        const additionalChildNode = <div id="additionalChild" />
        const wrapper = shallow(<TextField readonly additionalChild={additionalChildNode} />).dive()

        expect(wrapper.children().last().prop('readonly')).toBeTruthy()
    })

    test('Should throw "disabled" prop for text input if TextField is readonly', () => {
        const additionalChildNode = <div id="additionalChild" />
        const wrapper = shallow(<TextField readonly additionalChild={additionalChildNode} />).dive()

        expect(wrapper.children().first().prop('disabled')).not.toBeTruthy()
    })

    test('Should throw "focused" prop for wrapper when input was clicked', () => {
        const wrapper = shallow(<TextField size="md" />).dive()

        expect(wrapper.prop('focused')).toBeFalsy()

        wrapper.children().first()
            .simulate('click', {})

        expect(wrapper.prop('focused')).toBeTruthy()
    })

    test('Should throw formName for input', () => {
        const formNameValue = 'FormName'
        const wrapper = shallow(<TextField formName={formNameValue} />).dive()

        expect(wrapper.children().first().prop('form')).toBe(formNameValue)
    })

    test('Should throw placeholder for input', () => {
        const placeholderValue = 'PlaceholderValue'
        const wrapper = shallow(<TextField placeholder={placeholderValue} />).dive()

        expect(wrapper.children().first().prop('placeholder')).toBe(placeholderValue)
    })

    test('Should throw value for input', () => {
        const exampleValue = 'Cake is a lie!'
        const wrapper = shallow(<TextField value={exampleValue} />).dive()

        expect(wrapper.children().first().prop('value')).toBe(exampleValue)
    })

    test('Should correctly throw "focused" prop for wrapper when input was focused/unfocussed', () => {
        const wrapper = shallow(<TextField size="md" />).dive()

        expect(wrapper.prop('focused')).toBeFalsy()

        wrapper.children().first()
            .simulate('focus', {})

        expect(wrapper.prop('focused')).toBeTruthy()

        wrapper.children().first()
            .simulate('blur', {})

        expect(wrapper.prop('focused')).toBeFalsy()
    })

    test('Should correctly render button wrapper', () => {
        const a11yValues = { title: 'title' }
        const onClickMock = jest.fn()

        const wrapper1 = shallow(<TextField icon="icon" a11y={a11yValues} />).dive()

        expect(wrapper1.children().last().prop('as')).toBe('span')

        const wrapper2 = shallow(<TextField icon="icon" a11y={a11yValues} onClick={onClickMock} />).dive()

        expect(wrapper2.children().last().prop('as')).toBe('button')
    })

    test('Should call onFocus func when input was focused', () => {
        const onFocusMock = jest.fn()
        const wrapper = shallow(<TextField onFocus={onFocusMock} />).dive()

        expect(onFocusMock.mock.calls.length).toBe(0)

        wrapper.children().first()
            .simulate('focus', {})

        expect(onFocusMock.mock.calls.length).toBe(1)
    })

    test('Should call onBlur func when input was unfocussed', () => {
        const onBlurMock = jest.fn()
        const wrapper = shallow(<TextField onBlur={onBlurMock} />).dive()

        expect(onBlurMock.mock.calls.length).toBe(0)

        wrapper.children().first()
            .simulate('blur', {})

        expect(onBlurMock.mock.calls.length).toBe(1)
    })

    test('Should call onChange func when input was changed', () => {
        const onChangeMock = jest.fn()
        const wrapper = shallow(<TextField onChange={onChangeMock} />).dive()

        expect(onChangeMock.mock.calls.length).toBe(0)

        wrapper.children().first()
            .simulate('change', {})

        expect(onChangeMock.mock.calls.length).toBe(1)
    })

    test('Should\'nt call onChange func when input is disabled', () => {
        const onChangeMock = jest.fn()
        const wrapper = shallow(<TextField disabled onChange={onChangeMock} />).dive()

        expect(onChangeMock.mock.calls.length).toBe(0)

        wrapper.children().first()
            .simulate('change', {})

        expect(onChangeMock.mock.calls.length).toBe(0)
    })

    test('Should call onClick func if button was clicked', () => {
        const a11yValues = { title: 'title' }
        const onClickMock = jest.fn()
        const wrapper = shallow(<TextField icon="icon" a11y={a11yValues} onClick={onClickMock} />).dive()

        expect(onClickMock.mock.calls.length).toBe(0)

        wrapper.children().last()
            .simulate('click', {})

        expect(onClickMock.mock.calls.length).toBe(1)
    })
})
