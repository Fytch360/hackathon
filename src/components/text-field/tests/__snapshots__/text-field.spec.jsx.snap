// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> Snapshot 1`] = `
.emotion-0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  border: solid 1px;
  border-radius: 4px;
  -webkit-transition: 0.17s ease box-shadow;
  transition: 0.17s ease box-shadow;
  height: 54px;
  border-style: solid;
  border: solid 1px;
  pointer-events: none;
  padding: 0;
  padding-right: 16px;
  padding-left: 16px;
}

.emotion-0:hover {
  border: solid 1px;
}

.emotion-2 {
  -webkit-tap-highlight-color: transparent;
  border-radius: 4px;
  width: 100%;
  outline: none;
  display: block;
  font-weight: 500;
  background-color: transparent;
  -webkit-appearance: textfield;
  -moz-appearance: textfield;
  -ms-appearance: textfield;
  appearance: textfield;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  margin: 0;
  margin-right: 0px;
  font-size: 15px;
  line-height: 20px;
}

.emotion-2::-ms-clear {
  display: none;
}

.emotion-2::-webkit-inner-spin-button,
.emotion-2::-webkit-outer-spin-button,
.emotion-2::-webkit-clear-button,
.emotion-2::-webkit-inner-spin-button,
.emotion-2::-webkit-calendar-picker-indicator {
  display: none;
}

.emotion-2:-webkit-autofill,
.emotion-2:-webkit-autofill:hover,
.emotion-2:-webkit-autofill:focus,
.emotion-2:-webkit-autofill:active {
  -webkit-text-fill-color: inherit;
  box-shadow: 0 0 0 1000px transparent inset;
  -webkit-transition: background-color 5000s ease-in-out 0s;
  transition: background-color 5000s ease-in-out 0s;
}

.emotion-4 {
  padding: 0;
  margin-left: 20px;
  cursor: pointer;
  border-radius: 4px;
  outline: none;
  width: 36px;
  height: 36px;
  position: relative;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: background-color 0.17s ease-in-out,box-shadow 0.17s ease-in-out,border-color 0.17s ease-in-out;
  transition: background-color 0.17s ease-in-out,box-shadow 0.17s ease-in-out,border-color 0.17s ease-in-out;
}

.emotion-4::before {
  position: absolute;
  height: 36px;
  width: 1px;
  content: '';
  right: 48px;
}

body:not(.pointer-events) .emotion-4:focus {
  box-shadow: inset 0px 0px 0px 2px;
}

.emotion-6 {
  width: 36px;
  height: 36px;
  overflow: initial;
}

.emotion-11 {
  display: inline-block;
  position: relative;
  width: 20px;
  height: 20px;
  overflow: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 36px;
  height: 36px;
  overflow: initial;
}

.emotion-11 svg {
  display: block;
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%, -50%);
  -moz-transform: translate(-50%, -50%);
  -ms-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
}

.emotion-18 {
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: middle;
  display: inline-block;
  position: relative;
  width: 20px;
  height: 20px;
  overflow: hidden;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 36px;
  height: 36px;
  overflow: initial;
}

.emotion-18 svg {
  display: block;
}

.emotion-18 svg {
  display: block;
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%, -50%);
  -moz-transform: translate(-50%, -50%);
  -ms-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
}

<OmittedComponent
  a11y={
    Object {
      "label": "Демонстрационное поле",
    }
  }
  disabled={true}
  icon="iconExampleValue"
  id="idValue"
  onChange={[MockFunction]}
  onClick={[MockFunction]}
  placeholder="Текст"
  value="testValue"
>
  <Component
    a11y={
      Object {
        "label": "Демонстрационное поле",
      }
    }
    disabled={true}
    icon="iconExampleValue"
    id="idValue"
    onChange={[MockFunction]}
    onClick={[MockFunction]}
    placeholder="Текст"
    value="testValue"
  >
    <WrapperStyled
      disabled={true}
      focused={false}
      horizontalPadding="inner"
      size="md"
      verticalPadding="inner"
    >
      <span
        className="emotion-0 emotion-1"
        disabled={true}
        size="md"
      >
        <TextFieldStyled
          aria-label="Демонстрационное поле"
          disabled={true}
          horizontalMargin="zero"
          horizontalMarginDirection="right"
          horizontalPadding="inner"
          icon="iconExampleValue"
          id="idValue"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          placeholder="Текст"
          size="md"
          value="testValue"
          verticalPadding="inner"
        >
          <input
            aria-label="Демонстрационное поле"
            className="emotion-2 emotion-3"
            disabled={true}
            id="idValue"
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            placeholder="Текст"
            size="md"
            value="testValue"
          />
        </TextFieldStyled>
        <WrapperButtonStyled
          as="span"
          bordered={true}
          onClick={[MockFunction]}
          size="md"
        >
          <span
            className="emotion-4 emotion-5"
            onClick={[MockFunction]}
            size="md"
          >
            <IconStyled
              icon="iconExampleValue"
              size="md"
            >
              <IconViewBox
                className="emotion-6 emotion-7"
                icon="iconExampleValue"
                size="md"
              >
                <IconViewBoxStyled
                  className="emotion-6 emotion-7"
                  icon="iconExampleValue"
                  size="md"
                >
                  <Icon
                    className="emotion-7 emotion-11 emotion-12"
                    icon="iconExampleValue"
                    size="md"
                  >
                    <WrapperStyled
                      className="emotion-7 emotion-11 emotion-12"
                      dangerouslySetInnerHTML={
                        Object {
                          "__html": "iconExampleValue",
                        }
                      }
                      onClick={[Function]}
                    >
                      <span
                        className="emotion-7 emotion-12 emotion-18 emotion-19"
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "iconExampleValue",
                          }
                        }
                        onClick={[Function]}
                      />
                    </WrapperStyled>
                  </Icon>
                </IconViewBoxStyled>
              </IconViewBox>
            </IconStyled>
          </span>
        </WrapperButtonStyled>
      </span>
    </WrapperStyled>
  </Component>
</OmittedComponent>
`;
